src/app/cli/src/cli_entrypoint/mina_cli_entrypoint.ml

```diff
@@ -369,6 +377,14 @@ let setup_daemon logger =
       ~doc:
         "true|false whether to track the set of all peers ever seen for the \
          all_peers metric (default: false)"
+  and uptime_url_string =
+    flag "--uptime-url" ~aliases:["uptime-url"] (optional string)
+      ~doc:"URL URL of the uptime service of the Mina delegation program"
+  and uptime_submitter_string =
+    flag "--uptime-submitter" ~aliases:["uptime-submitter"] (optional string)
```

Comment:
Why is this needed / allowed to be different than the --block-producer-key? 
Separately, if the snark work is submitted, what address is used for the snark work?

Reply:
I don't why a separate submitter was given as a requirement. It's the submitter that signs the payload.

Reply:
I think it also looks like the snark work uses this submitter key which will submit (and could get paid for) the snark work. Seems odd to me - I would expect snark work going to snarketplace to be from my snarkworker address, if specified. I guess if I am not running a snark worker at all ... it needs to go somewhere? Or perhaps should not be submitted to the snarketplace if I don't have a snark coordinator or snark worker and a snark fee. 🤷 seems odd.

Reply:
A daemon can stake for multiple block producer keys at a time. Therefore, there is no 1-1 relationship to a node and a block producer key. This disambiguates which block producer keypair should be used for the delegation uptime submission, since otherwise we may not know which one to use.

Reply:
But why not just use the existing --block-producer-key parameter? What reason is there for them to be different values?